### ISSUE TITLE
The unique index constraint was not updated after adding a new field.

### ISSUE THREAD

**Issue by guilhermeneves** [ariga/atlas#3092]
Hello,

We are facing an issue when trying to insert a new column into an existing unique index constraint. Although Atlas correctly identifies the changes, it does not apply a delete followed by a recreate (which is what is expected if it were handled manually). Instead, it simply ignores the changes but continues to identify them with each new deployment, only applying the other changes.

Unique Index Constraint Changes Identified:
![image](https://github.com/user-attachments/assets/4fdce5cc-2cf2-4c36-9565-f9f6ce4184d7)

Changes applied by Atlas:
![image](https://github.com/user-attachments/assets/74760810-2f3f-47da-8381-19199c0c22ec)

**Comment by a8m** [comment]
Hey! 

In order to help with this, please write here the Atlas version you are using and add a small example that reproduces this on the latest version.

**Comment by guilhermeneves** [comment]
I'm using atlas version 0.25

Here a small example to reproduce the error, as we're using terraform just a SQL DDL changes is necessary to trigger a change:

1. Create a simple table with unique INDEX
```
CREATE TABLE [simple_table] (
    [id] INT PRIMARY KEY,
    [column1] VARCHAR(50),
    [column2] VARCHAR(50),
    [column3] INT
);

CREATE UNIQUE NONCLUSTERED INDEX [idx_unique_x] ON [simple_table](column1 ASC, column2 ASC);
```

2. Change the SQL statement to include a new field in the UNIQUE INDEX:

```
CREATE TABLE [simple_table] (
    [id] INT PRIMARY KEY,
    [column1] VARCHAR(50),
    [column2] VARCHAR(50),
    [column3] INT
);

CREATE UNIQUE NONCLUSTERED INDEX [idx_unique_x] ON [simple_table](column1 ASC, column2 ASC, column3 ASC);
```

**Comment by a8m** [comment]
Can you please paste the full version? Also, do you also get this on latest version?

cc @giautm - this looks like SQL Server driver issue.

**Comment by guilhermeneves** [comment]
I didn't try in the last version, I will try it.

This is the real full table version that using Atlas:

**Original Version:**
```
-- Create "tbl_order_recommendations" table
CREATE TABLE [dbo].[tbl_order_recommendations] (
    [id] int IDENTITY (1, 1) NOT NULL,
    [customer_sku_id] varchar(40) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
    [customer_store_id] int NOT NULL,
    [customer_vendor_id] varchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
    [placement_date] date NOT NULL,
    [delivery_date] date NOT NULL, -- NOTE: "delivery date using observable vendor real leadtimes"
    [sku_uom_qty] float NOT NULL,
    [group_uom_qty] float NOT NULL,
    [pack_qty] int NOT NULL,
    [sku_to_group_uom_conversion_factor] float NOT NULL,
    [pack_size] float NULL,
    [customer_delivery_date] date NULL, -- FIXME: "DEPRECATED, new field: sla_date."
    [sla_date] date DEFAULT ('1970-01-01') NOT NULL,
    [active] bit DEFAULT (0) NOT NULL,
    CONSTRAINT [PK_tbl_order_recommendations] PRIMARY KEY CLUSTERED ([id] ASC),
    CONSTRAINT [FK_recommendationsku] FOREIGN KEY ([customer_sku_id]) REFERENCES [dbo].[tbl_sku] ([customer_sku_id]) ON UPDATE NO ACTION ON DELETE NO ACTION,
    CONSTRAINT [FK_recommendationstore] FOREIGN KEY ([customer_store_id]) REFERENCES [dbo].[tbl_store] ([customer_store_id]) ON UPDATE NO ACTION ON DELETE NO ACTION,
    CONSTRAINT [FK_recommendationvendor] FOREIGN KEY ([customer_vendor_id]) REFERENCES [dbo].[tbl_vendor] ([customer_vendor_id]) ON UPDATE NO ACTION ON DELETE NO ACTION
);

-- Create index "UQ_tbl_order_recommendations_natural_comp_key" to table: "tbl_order_recommendations"
CREATE UNIQUE NONCLUSTERED INDEX [UQ_tbl_order_recommendations_natural_comp_key] ON [dbo].[tbl_order_recommendations] (
    [customer_sku_id] ASC,
    [customer_store_id] ASC,
    [customer_vendor_id] ASC,
    [placement_date] ASC,
    [delivery_date] ASC
);
```
**Changed Version:**

```
-- Create "tbl_order_recommendations" table
CREATE TABLE [dbo].[tbl_order_recommendations] (
    [id] int IDENTITY (1, 1) NOT NULL,
    [customer_sku_id] varchar(40) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
    [customer_store_id] int NOT NULL,
    [customer_vendor_id] varchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
    [placement_date] date NOT NULL,
    [delivery_date] date NOT NULL, -- NOTE: "delivery date using observable vendor real leadtimes"
    [sku_uom_qty] float NOT NULL,
    [group_uom_qty] float NOT NULL,
    [pack_qty] int NOT NULL,
    [sku_to_group_uom_conversion_factor] float NOT NULL,
    [pack_size] float NULL,
    [customer_delivery_date] date NULL, -- FIXME: "DEPRECATED, new field: sla_date."
    [sla_date] date DEFAULT ('1970-01-01') NOT NULL,
    [active] bit DEFAULT (0) NOT NULL,
    [timestamp_run] datetime DEFAULT ('1970-01-01 00:00:00') NOT NULL,
    CONSTRAINT [PK_tbl_order_recommendations] PRIMARY KEY CLUSTERED ([id] ASC),
    CONSTRAINT [FK_recommendationsku] FOREIGN KEY ([customer_sku_id]) REFERENCES [dbo].[tbl_sku] ([customer_sku_id]) ON UPDATE NO ACTION ON DELETE NO ACTION,
    CONSTRAINT [FK_recommendationstore] FOREIGN KEY ([customer_store_id]) REFERENCES [dbo].[tbl_store] ([customer_store_id]) ON UPDATE NO ACTION ON DELETE NO ACTION,
    CONSTRAINT [FK_recommendationvendor] FOREIGN KEY ([customer_vendor_id]) REFERENCES [dbo].[tbl_vendor] ([customer_vendor_id]) ON UPDATE NO ACTION ON DELETE NO ACTION
);

-- Create index "UQ_tbl_order_recommendations_natural_comp_key" to table: "tbl_order_recommendations"
CREATE UNIQUE NONCLUSTERED INDEX [UQ_tbl_order_recommendations_natural_comp_key] ON [dbo].[tbl_order_recommendations] (
    [customer_sku_id] ASC,
    [customer_store_id] ASC,
    [customer_vendor_id] ASC,
    [placement_date] ASC,
    [delivery_date] ASC,
    [timestamp_run] ASC
);
```

**Comment by giautm** [comment]
Hello @guilhermeneves, thank you for report this issue. We have fixed it and release in the version v0.27.1-da7a42c-canary,

Please upgrade and give a try. Thank you,

<img width="842" alt="Screenshot 2024-09-04 at 12 56 27" src="https://github.com/user-attachments/assets/137afd7e-b446-4bf6-966c-e55ff9a505f4">
